python/mujinwebstackclient/controllerwebclientraw.py

```diff
@@ -248,3 +249,19 @@ def CallGraphAPI(self, query, variables=None, headers=None, timeout=5.0):
             raise ControllerGraphClientException(_('Unexpected server response %d: %s') % (statusCode, raw), statusCode=statusCode, response=response)
 
         return content['data']
+
+    def Login(self, timeout=5):
```

Comment:
@LLipter Can we implement a custom auth class derived from `AuthBase` and use it for `self._session.auth` instead?

Reply:
The current method is going to make a login request even if no API calls are made using the client, which is wasteful

Reply:
@woswos Fixed. Could you have a look again?

Reply:
@LLipter I think we can handle both JWT and basic auth login entirely in the custom auth class without the login functions to further simplify the design.

Reply:
done